lib/improver/tests/metadata/test_utilities.py

```diff
@@ -64,38 +68,94 @@ def test_basic(self):
         self.assertSequenceEqual(result.coords(dim_coords=False),
                                  self.template_cube.coords(dim_coords=False))
         self.assertFalse(np.allclose(result.data, self.template_cube.data))
-        self.assertFalse(result.attributes)
+        self.assertDictEqual(result.attributes, self.mandatory_attributes)
         self.assertFalse(result.cell_methods)
         self.assertEqual(result.data.dtype, np.float32)
 
     def test_attributes(self):
-        """Test attributes can be set on the output cube"""
-        attributes = {"source": "IMPROVER"}
+        """Test optional attributes can be set on the output cube, and override
+        the values in mandatory_attributes"""
+        attributes = {"source": "Mars", "mosg__model_configuration": "uk_det"}
+        expected_attributes = self.mandatory_attributes
+        for attr in attributes:
```

Comment:
Could just do a dictionary.update (but not bothered if you change it or not)

Reply:
I like things to be tidy - will fix.